tools/licenses/lib/licenses.dart

```diff
@@ -116,21 +153,132 @@ LicenseType convertBodyToType(String body) {
 }
 
 abstract class LicenseSource {
+  String get name;
+  String get libraryName;
+  String get officialSourceLocation;
   List<License>? nearestLicensesFor(String name);
   License? nearestLicenseOfType(LicenseType type);
   License? nearestLicenseWithName(String name, { String? authors });
 }
 
-abstract class License implements Comparable<License> {
+class Assignment {
+  const Assignment(this.license, this.target, this.source);
+  final License license;
+  final String target;
+  final LicenseSource source;
+}
+
+class GroupedLicense {
+  GroupedLicense(this.type, this.body);
+  final LicenseType type;
+  final String body;
+  final Set<String> targets = <String>{};
+  final Set<String> libraries = <String>{};
+  final Set<String> origins = <String>{};
```

Comment:
These three could also use docstrings, since they aren't necessarily obvious in meaning.

Reply:
done